### PR TITLE
chore: reset options for `machine_configuration_apply` resoure.

### PR DESCRIPTION
**Pull request by frezbo**
Support optionally resetting machines on destroy for `talos_machine_configuration_apply` resource.

Fixes: #133

### CODE REVIEW

File: docs/resources/machine_configuration_apply.md [hunk 1]
```diff
@@ -73,11 +77,22 @@ Required:
...
 
 
+<a id="nestedatt--on_destroy"></a>
+### Nested Schema for `on_destroy`
+
+Optional:
+
+- `graceful` (Boolean) Graceful indicates whether node should leave etcd before the upgrade, it also enforces etcd checks before leaving. Default true
+- `reboot` (Boolean) Reboot indicates whether node should reboot or halt after resetting. Default false
+- `reset` (Boolean) Reset the machine to the initial state (STATE and EPHEMERAL will be wiped). Default false
```
Comment (smira) [inline docs/resources/machine_configuration_apply.md:87]: this is a bit confusing... if this is false, it doesn't reset?

this looks same level as `graceful`

Comment (smira) [reply docs/resources/machine_configuration_apply.md:87]: I mean it could be `enabled` or something like that?

Comment (frezbo) [reply docs/resources/machine_configuration_apply.md:87]: it's the default value the provider sets, the user can ovveride

**Pull request merged**